keps/sig-node/127-user-namespaces/README.md

```diff
@@ -316,121 +325,121 @@ The picked range will be stored under a file named `userns` in the pod folder
 (by default it is usually located in `/var/lib/kubelet/pods/$POD/userns`). This
 way, the Kubelet can read all the allocated mappings if it restarts.
 
-During phase 1, to make sure we don't exhaust the host UID namespace, we will
+During alpha, to make sure we don't exhaust the host UID namespace, we will
 limit the number of pods using user namespaces to `min(maxPods, 1024)`. This
 leaves us plenty of host UID space free and this limits is probably never hit in
 practice. See UNRESOLVED for more some UNRESOLVED info we still have on this.
 
-##### pkg/volume changes for phase I
+### Handling of stateless volumes
 
-We need the files created by the kubelet for these volume types to be readable
-by the user the pod is mapped to in the host. Luckily, Kuberentes already
-supports specifying an [fsUser and fsGroup for volumes][volume-mounter-args] for
-projected SA tokens so we just need to extend that to the other volume types we
-support in phase I.
+Only the aforementioned volume types are supported. If other volume types are
+used, a clear error is thrown during API validation of the pod.spec.
 
-First, we need to teach the [operation executor][operation-executor] to convert
-the container UIDs/GIDs to the host UIDs/GIDs that this pod is mapped to. To do
-this, we will modify the [volumeHost interface][volumeHost-interface] adding
-functions to transform a container ID to the corresponding host ID, so the
-operation executor will just call that function (via
-self.volumePluginMgr.Host.GetHostIDsForPod())
+When the volumes used are supported, the kubelet will set the `uid_mappings` and
+`gid_mappings` in the CRI `Mount message`. It will use the same mappings the
+kubelet allocated for the user namespace (see previous section for more details
+on this).
 
-This function will call the kubelet, which will use the user namespace manager
-created to transform the IDs for that pod. The [volumeHost
-interface][volumeHost-interface] will then have this new method:
+The container runtime, when it receives the `Mount message` with the UID/GID
+mappings fields set, it uses the Linux idmap mounts support to create such a
+mount. If the kernel/filesystem used doesn't support idmap mounts, a clear error
+will be thrown by the OCI runtime at pod creation.
```

Comment:
Is there a fallback to overlay metadata chown possible if idmaps support isn't present?

Reply:
I think we were trying to add idmaps mount for solving issues with specific files but I don't see those called out in this KEP update.

Reply:
@mrunalp As we talked, the second comment is addressed a few lines after this one.

Overlay is out of the picture here. Yes, a full chown of the rootfs can be done, or a chown with metacopy (so only metadata is copied, not the whole file) is completely possible by the container runtime. This section covers only "volumes" (configmap, secrets, bind-mounts of files like `/etc/resol.conf`, etc.).

This doesn't affect the rootfs and container runtimes can do what they please. In fact, containerd so far is not using idmap mounts nor metacopy for the rootfs (PRs are open and in progress to do that, though).